### PR TITLE
support new item in ScanContext

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #5926 

Problem Summary:

### What is changed and how it works?
Add new item in ScanContext, to record the regions read in local streams and remote streams

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Transaction/LearnerRead.cpp [hunk 1]
```diff
@@ -475,12 +475,14 @@ void validateQueryInfo(
             iter == regions_snapshot.end() || iter->second != region)
         {
             status = RegionException::RegionReadStatus::NOT_FOUND;
+            mvcc_query_info.scan_context->total_local_region_num--;
```
Comment (SeaRise) [inline dbms/src/Storages/Transaction/LearnerRead.cpp:478]: I prefer to handle in `buildLocalStreamsForPhysicalTable`.
```
catch (RegionException & e)
{
    mvcc_query_info.scan_context->total_local_region_num -= e.unavailable_region;
}
```

**Review by SeaRise — APPROVED** [review]
lgtm

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**